src/ring_theory/ideal/basic.lean

```diff
@@ -490,16 +553,34 @@ namespace ideal
 /-- All ideals in a division ring are trivial. -/
 lemma eq_bot_or_top : I = ⊥ ∨ I = ⊤ :=
 begin
-  rw or_iff_not_imp_right,
-  change _ ≠ _ → _,
-  rw ideal.ne_top_iff_one,
+  rw [or_iff_not_imp_right, ← ne.def, ideal.ne_top_iff_one],
   intro h1,
   rw eq_bot_iff,
   intros r hr,
   by_cases H : r = 0, {simpa},
   simpa [H, h1] using I.mul_mem_left r⁻¹ hr,
 end
 
+@[priority 100] -- see Note [lower instance priority]
+noncomputable instance : fintype $ ideal K :=
```

Comment:
What makes this noncomputable?

Reply:
I'm using `classical.prop_decidable` to supply the typeclass `[decidable_eq $ right_ideal K]` which is used in the line `elems := {⊥, ⊤}`.

Reply:
I think you can avoid needing that if you use `finset.cons`

Reply:
That is
```lean
{ elems := finset.cons ⊥ {⊤} $ finset.mem_singleton.not.mpr bot_ne_top,
  complete := λ I, by simp [eq_bot_or_top] }
```

Reply:
Since it follows trivially, you could add the lemma
```lean
@[simp] lemma card_ideal : fintype.card (ideal K) = 2 := rfl
```
too

Reply:
Now done!